backend/apps/ifc_validation/checks/check_bsdd.py

```diff
@@ -1,396 +1,622 @@
+
+            assignment_result = { 
+                "assignment_id": rel.id(),
+                "assignment_name": rel.Name,
+                "assignment_description": rel.Description,
+                "assignment_rel_objects": [],
+                "assignment_rel_class": {
+                    "id": rel.RelatingClassification.id(),
+                    "type": rel.RelatingClassification.is_a(),
+                }
+            }
+            for o in rel.RelatedObjects:
+                assignment_result["assignment_rel_objects"] += [{
+                    "id": o.id(),
+                    "type": o.is_a()
+                }]
+
+            bsdd_results['assignments'] += [assignment_result]
+
+    # bSDD properties
+    # https://github.com/buildingSMART/bSDD/blob/master/Documentation/bSDD-IFC%20documentation.md#4-bsdd-properties
+    if len(ifc_file_properties):
+
+        for _, prop in enumerate(ifc_file_properties):
+
+            logger.debug(f"File '{file_name}' for task_id = {task_id} has IfcProperty {prop.Name}.")
+
+            property_result = { 
+                "property_id": prop.id(),
+                "property_type": prop.is_a(),
+                "property_name": prop.Name,
+                "property_identifier": first_available_attr(prop, ['Specification', 'Description']), # resp IFC 4.3 - IFC4/IFC 2x3
+                "property_predefined_value": None,
+                "property_predefined_type": None,
+                "property_unit": prop.Unit if hasattr(prop, 'Unit') else None,
+                "property_allowed_values": None,
+            }
+
+            # NominalValue
+            if hasattr(prop, 'NominalValue') and prop.NominalValue:
+                value = prop.NominalValue
+                property_result["property_predefined_value"] = value.wrappedValue
+                property_result["property_predefined_type"] = value.is_a()                
+
+            # EnumerationValues
+            if hasattr(prop, 'EnumerationValues') and prop.EnumerationValues:
+                property_result["property_allowed_values"] = []
+                for enum_value in prop.EnumerationValues:
+                    if hasattr(enum_value, 'wrappedValue') and enum_value.wrappedValue:
+                        property_result["property_allowed_values"] += [{
+                            'value': enum_value.wrappedValue,
+                            'type': enum_value.is_a()
+                        }]
+                    elif enum_value:
+                        property_result["property_allowed_values"] += [{
+                            'value': str(enum_value),
+                            'type': str(type(enum_value))
+                        }]
+
+            # Unit
+            if hasattr(prop, 'Unit') and prop.Unit and isinstance(prop.Unit, ifcopenshell.entity_instance):
+                property_result["property_unit"] = prop.Unit.__str__()
```

Comment:
`property_unit` appears to never be checked, but here it's a str, above it's an entity_instance.